### PR TITLE
Revising some stuff in project controller

### PR DESCRIPTION
**Pull request by pellaea**
Project add_members was listing all verified users -- *way* too many.  Limited it to the last 100 to login, and added a write-in autocompleted field.

Added flash notices for both add_members and change_status.

(Added autocomplete and flash notices for Switch Users, too.)

Should be pretty by-the-numbers, review probably not necessary, but welcome if anyone has time.

### CODE REVIEW

File: app/controllers/project_controller.rb [hunk 1]
```diff
@@ -294,17 +294,29 @@ def admin_request
...
     pass_query_params
-    if @project = find_or_goto_index(Project, params[:id].to_s)
-      @users = User.where.not(verified: nil).order("login, name").to_a
-      if !@project.is_admin?(@user)
-        redirect_with_query(action: "show_project", id: @project.id)
-      elsif params[:candidate].present?
-        @candidate = User.find(params[:candidate])
-        set_status(@project, :member, @candidate, :add)
-      end
+    return unless @project = find_or_goto_index(Project, params[:id].to_s)
```
Comment (JoeCohen) [inline app/controllers/project_controller.rb:297]: Sorry I didn't notice this earlier
```suggestion
    return unless (@project = find_or_goto_index(Project, params[:id].to_s))
```

File: app/controllers/project_controller.rb [hunk 2]
```diff
@@ -294,17 +294,29 @@ def admin_request
...
+    return must_be_project_admin!(@project.id) unless @project.is_admin?(@user)
+
+    @users = User.order("last_login desc").limit(100).to_a
+    @candidate = params[:candidate]
+    add_member(@candidate, @project) if @candidate.present?
```
Comment (JoeCohen) [inline app/controllers/project_controller.rb:302]:
```suggestion
    return unless (@candidate = params[:candidate]).present?
    
    add_member(@candidate, @project)
```

File: test/controllers/project_controller_test.rb [hunk 3]
```diff
@@ -403,6 +403,35 @@ def test_add_member_member
...
   end
 
+  def test_add_member_writein
+    eol_project = projects(:eol_project)
+    target_user = dick
+    assert_not(target_user.in_group?(eol_project.admin_group.name))
+    assert_not(target_user.in_group?(eol_project.user_group.name))
+    params = {
+      id: eol_project.id,
+      candidate: "dick <Tricky Dick>"
```
Comment (JoeCohen) [inline test/controllers/project_controller_test.rb:413]:
```suggestion
      candidate: "#{target_user.login} <#{target_user.name}>"
```
would make it easier to choose a different fixture if that were ever needed. I think also a bit less magical.

File: test/controllers/project_controller_test.rb [hunk 4]
```diff
@@ -403,6 +403,35 @@ def test_add_member_member
...
+    }
+    requires_login(:add_members, params, mary.login)
+    assert_response(:success)
+    target_user = User.find(target_user.id)
```
Comment (JoeCohen) [inline test/controllers/project_controller_test.rb:417]: Is this in order to reload `target_user`?

**Comment by JoeCohen** [comment]
Still looks good. Sorry that all I can offer is just picking around the edges. 
If that annoys you, please let me know.

File: test/controllers/project_controller_test.rb [hunk 5]
```diff
@@ -410,12 +410,12 @@ def test_add_member_writein
...
       id: eol_project.id,
-      candidate: "dick <Tricky Dick>"
+      candidate: "#{target_user.login} <Should Ignore This>"
     }
     requires_login(:add_members, params, mary.login)
     assert_response(:success)
-    target_user = User.find(target_user.id)
-    assert_not(target_user.in_group?(eol_project.admin_group.name))
+    target_user.reload
+    assert_not(target_user.reload.in_group?(eol_project.admin_group.name))
```
Comment (JoeCohen) [inline test/controllers/project_controller_test.rb:418]: Do we need to reload twice?
(I think you could get rid of line 417.)

Comment (pellaea) [reply test/controllers/project_controller_test.rb:418]: Oops!  Will fix that.

**Pull request merged**